src/AnalysisHelper.cc

```diff
@@ -13,1181 +14,1317 @@ using namespace std;
 
 AnalysisHelper::AnalysisHelper(string cfgname)
 {
-    TH1::SetDefaultSumw2(true);
+  TH1::SetDefaultSumw2(true);
 
-    nominal_name_ = "NOMINAL"; // used for nominal systematics
+  nominal_name_ = "NOMINAL"; // used for nominal systematics
 
-    nsplit_   = 1; // default: only 1 job
-    idxsplit_ = 0;
+  nsplit_   = 1; // default: only 1 job
+  idxsplit_ = 0;
 
-    cout << "@@ Parsing main config : " << cfgname << endl;
-    mainCfg_ = unique_ptr<CfgParser>(new CfgParser(cfgname));
-    bool success = readMainInfo();
-    if (!success)
-        cerr << "** AnalysisHelper : error : some information could not be retrieved from config" << endl;
+  cout << "@@ Parsing main config : " << cfgname << endl;
+  mainCfg_ = unique_ptr<CfgParser>(new CfgParser(cfgname));
+  bool success = readMainInfo();
+  if (!success) {
+    cerr << "** AnalysisHelper : error : some information could not be retrieved from config" << endl;
+    std::exit(1);
+  }
 }
 
 AnalysisHelper::~AnalysisHelper()
 {}
 
 bool AnalysisHelper::readMainInfo()
 {
-    if (!mainCfg_->hasOpt("configs::cutCfg") || !mainCfg_->hasOpt("configs::sampleCfg"))
+  if (!mainCfg_->hasOpt("configs::cutCfg") || !mainCfg_->hasOpt("configs::sampleCfg"))
+  {
+    cerr << "** AnalysisHelper : error : configs for sample and cut not provided in main cfg" << endl;
+    return false;
+  }
+  string sampleCfgName = mainCfg_->readStringOpt("configs::sampleCfg");
+  string cutCfgName = mainCfg_->readStringOpt("configs::cutCfg");
+  cout << "@@ sample cfg          : " << sampleCfgName << endl;
+  cout << "@@ selection cfg       : " << cutCfgName << endl;
+
+  cutCfg_ = unique_ptr<CfgParser>(new CfgParser(cutCfgName));
+  sampleCfg_ = unique_ptr<CfgParser>(new CfgParser(sampleCfgName));
+
+  if (!(mainCfg_->hasOpt("general::lumi"))) return false;
+  lumi_ = mainCfg_->readFloatOpt("general::lumi");
+  cout << "@@ lumi                : " << lumi_ << endl;
+
+  outputFileName_ = "outPlotter.root"; // override default only if specified
+  if (mainCfg_->hasOpt("general::outputFileName"))
+    outputFileName_ = mainCfg_->readStringOpt("general::outputFileName");
+  cout << "@@ output file  name   : " << outputFileName_<< endl;
+
+  if (mainCfg_->hasSect("merge_plots"))
```

Comment:
Hi @bfonta ! I see that you changed this flag from "merge" to "merge_plots". I was not aware of this. Can you please change the mainCfg of 2018 accordingly so that people can have a look in case of doubts? Is there any other change like this we should all be aware of? Thanks :)

Reply:
Follow up: I believe this change require some modifications in script/combineFillerOutputs.py or scripts/modules/OutputManager.py to work properly.

Reply:
@dzuolo Well spotted ;)
That is still work in progress, but the idea is to avoid running the histogramming step twice. For the non-resonant analysis we had to run it once for plotting the distributions and once for obtaining the limits (we do not merge the less-important backgrounds in ```others``` when fitting). I will ping this thread once this additional step is working.

Reply:
The idea is quite nice! Thanks for the clarification!